### PR TITLE
Add RiseFM to PreMiD

### PR DESCRIPTION
**Pull request by DawsonCodes**
image
Proof of the presence working.
Presence for: RiseFM

### CODE REVIEW

File: websites/R/RiseFM/presence.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+let rname: string, rartist: string, listeners: string, islive: string, presenter: string, lTitle: string, yes: boolean = false;
+
+function metadataListener(): void {
+  const data = JSON.parse(this.responseText);
+  rname = data.now.title;
+  rartist = data.now.artist;
+  listeners = data.listeners.current;
+  islive = data.live.autoDJ;
+  presenter = data.live.userrname;
```
Comment (ririxichan) [inline websites/R/RiseFM/presence.ts:14]: wouldn't it be better solution to just use const?

File: websites/R/RiseFM/presence.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+let ltStart = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+
+const toggleelapse = await presence.getSetting("elapse");
+const details = await presence.getSetting("details");
+const state = await presence.getSetting("state");
+const small = await presence.getSetting("changesmalltext");
+
+  const presenceData: PresenceData = {
```
Comment (ririxichan) [inline websites/R/RiseFM/presence.ts:34]: merge consts

**Review by ririxichan — CHANGES_REQUESTED** [review]

**Review by ririxichan — APPROVED** [review]

File: websites/R/RiseFM/presence.ts (hunk 1 above)
Comment (Bas950) [reply websites/R/RiseFM/presence.ts:13]: No he needs let here because he is setting it in a function and using it outside of the function

**Review by Bas950 — DISMISSED** [review]

File: websites/R/RiseFM/presence.ts (hunk 1 above)
Comment (ririxichan) [reply websites/R/RiseFM/presence.ts:13]: right but he changed now things that SHOULD be const and merged

File: websites/R/RiseFM/presence.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+function doMeta(): void {
+  const xhttp = new XMLHttpRequest();
+  xhttp.addEventListener("load", metadataListener);
+  xhttp.open("GET", "https://api.risefm.pw/stats", true);
+  xhttp.send();
+}
+
+setInterval(doMeta, 10000);
+let ltStart = Math.floor(Date.now() / 1000);
```
Comment (ririxichan) [inline websites/R/RiseFM/presence.ts:24]: let -> const again

File: websites/R/RiseFM/presence.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+presence.on("UpdateData", async () => {
+
+const toggleelapse = await presence.getSetting("elapse"),
+details = await presence.getSetting("details"),
+state = await presence.getSetting("state"),
+small = await presence.getSetting("changesmalltext");
+
+  const presenceData: PresenceData = {
```
Comment (ririxichan) [inline websites/R/RiseFM/presence.ts:33]: merge consts....

File: websites/R/RiseFM/presence.ts [hunk 5]
```diff
@@ -0,0 +1,62 @@
+const presence = new Presence({
+  clientId: "773701779434897419"
+});
+
+let rname: string, rartist: string, listeners: string, islive: string, presenter: string, lTitle: string, yes: boolean = false;
```
Comment (ririxichan) [inline websites/R/RiseFM/presence.ts:5]: also, where you use yes variable? Because it is not even used.

**Review by ririxichan — CHANGES_REQUESTED** [review]

**Comment by ririxichan** [comment]
Site is not 2 months old. Feel free to do pr when it will have minimum.

**Pull request closed without merging**